rtgui/batchqueue.cc

```diff
@@ -1022,20 +1021,19 @@ Glib::ustring BatchQueue::calcAutoFileNameBase (const Glib::ustring& origFileNam
                     // Insert formatted date/time value. Character after 't' defines time source
                     if (++ix < options.savePathTemplate.size()) {
                         Glib::DateTime dateTime;
-                        bool dateTimeIsValid = true;
                         switch(options.savePathTemplate[ix++])
                         {
                             case 'E':   // (approximate) time when export started
                                 dateTime = Glib::DateTime::create_now_local();
                                 break;
                             case 'F': // time when file was last saved
                             {
-                                struct stat fileStat;
-                                if (stat(origFileName.c_str(), &fileStat) == 0) {
-                                    time_t timestamp = fileStat.st_mtime;
-                                    dateTime = Glib::DateTime::create_now_local(timestamp);
-                                } else {
-                                    dateTimeIsValid = false; // file not found, access denied, etc.
+                                Glib::RefPtr<Gio::File> file = Gio::File::create_for_path(origFileName);
+                                if (file) {
+                                    Glib::RefPtr<Gio::FileInfo> info = file->query_info("time::modified");
```

Comment:
[`G_FILE_ATTRIBUTE_TIME_MODIFIED`](https://docs.gtk.org/gio/const.FILE_ATTRIBUTE_TIME_MODIFIED.html) is defined as `"time::modified"`.